src/routes/api/sections/querySection.ts

```diff
@@ -3,35 +3,51 @@ import sectionModel from '../../../models/section.model';
 import timeblockModel from '../../../models/timeblock.model';
 
 const querySection = async (req: Request, res: Response): Promise<void> => {
-    // Retrieves the room section with the id given in the link
-    const sectionInformation = await sectionModel.findOne({ _id: req.params.id }).catch((error) => {
-        res.status(404).json({ error });
-        return;
-    });
-    if (!sectionInformation) {
-        res.status(404);
-        return;
-    }
-
-    // Retrieves all time blocks for the room section within the time range given in the input
-    const sectionTimeblocks = await timeblockModel
-        .find({
-            sectionId: sectionInformation._id,
-            startsAt: { $gte: new Date(req.body.startsAt) },
-            endsAt: { $lte: new Date(req.body.endsAt) },
-        })
-        .catch((error) => {
+    const { id, startsAt, endsAt } = req.query;
+    if (id) {
+        // Retrieves the room section with the id given in the link
+        const sectionInformation = await sectionModel.findOne({ _id: id }).catch((error) => {
             res.status(404).json({ error });
             return;
         });
+        if (!sectionInformation) {
+            res.status(404);
+            return;
+        }
+
+        if (startsAt && endsAt) {
+            if (typeof startsAt != 'string' || typeof endsAt != 'string') {
+                res.status(400).json({ error: 'validation failed, variable types are incorrect' });
+                return;
+            } else {
+                // Retrieves all time blocks for the room section within the time range given in the input
+                const sectionTimeblocks = await timeblockModel
+                    .find({
+                        sectionId: sectionInformation._id,
```

Comment:
I wonder if we need _id would be needed for query since we already have start and end date for the section, if _id is added it is only querying one section. The api doc also didn't indicate _id.